python/fledge/services/core/server.py

```diff
@@ -1120,10 +1125,17 @@ async def register(cls, request):
             if not isinstance(service_management_port, int):
                 raise web.HTTPBadRequest(reason='Service management port can be a positive integer only')
 
+            # If token then check single use token verification; if bad then return 4XX
             if token is not None:
                 if not isinstance(token, str):
                     msg = 'Token can be a string only'
-                    raise web.HTTPBadRequest(reason=msg, body=json.dumps(msg))
+                    raise web.HTTPBadRequest(reason=msg)
+
+                # Check startup token exists
+                if ServiceRegistry.checkStartupToken(service_name, token) == False:
```

Comment:
@pintomax We must check if a startup token was issued and while registering then token is not present in payload. 

if ServiceRegistry.getStartupToken then ensure token is not data.get('token', None) and match. Here we are relying only if supplied by consumer service. and allow registration with None. The None is allowed only for those which were not issued any startup token.

Reply:
@praveen-garg is that resolved by your last commit?

Reply:
nope

Reply:
Would you mind adding that check while I'm testing all the connected services along with dispatcher?